plugins/crypto/ua_certificategroup_filestore.c

```diff
@@ -439,18 +439,10 @@ FileCertStore_createPkiDirectory(UA_CertificateGroup *certGroup, const UA_String
     char rootDirectory[PATH_MAX] = {0};
     size_t rootDirectorySize = 0;
 
-    /* Set base directory */
-    if(directory != NULL) {
-        if(directory->length >= PATH_MAX) {
-            return UA_STATUSCODE_BADINTERNALERROR;
-        }
-        memcpy(rootDirectory, directory->data, directory->length);
-    }
-    else {
-        if(getcwd(rootDirectory, PATH_MAX) == NULL) {
-            return UA_STATUSCODE_BADINTERNALERROR;
-        }
-    }
+    if(directory.length <= 0)
+        return UA_STATUSCODE_BADINTERNALERROR;
+
+    memcpy(rootDirectory, directory.data, directory.length);
```

Comment:
Potential buffer overflow here. You should ensure whether the length of the given directory does not exceed PATH_MAX.